src/theory/strings/output_channel.h

```diff
@@ -0,0 +1,308 @@
+/*********************                                                        */
+/*! \file output_channel.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Customized output channel for the theory of strings
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__STRINGS__OUTPUT_CHANNEL_H
+#define CVC4__THEORY__STRINGS__OUTPUT_CHANNEL_H
+
+#include "context/cdhashset.h"
+#include "theory/uf/equality_engine.h"
+
+#include <climits>
+#include <deque>
+
+namespace CVC4 {
+namespace theory {
+namespace strings {
+
+class TheoryStrings;
+
+/** Output channel strings
+ *
+ * The purpose of this class is to process inference steps for strategies
+ * in the theory of strings.
+ *
+ * In particular, inferences are given to this class via calls to functions:
+ *
+ * sendInternalInference, sendInference, sendSplit
+ *
+ * This class decides how these calls will be processed, where the main design
+ * decisions it considers are whether the conclusions corresponding to these
+ * calls will be processed as:
+ *
+ * (1) Internally in the strings solver, via calls to the equality engine. We
+ * refer to these literals as "facts",
```

Comment:
What I call "facts" does refer to literals. 

A "conclusion" indeed may be an arbitrary formula, which can be processed to multiple literal "facts".

Reply:
Right. In the context, it is still a bit confusing IMHO because the text says: 

> whether the _conclusions_ will be processed [...] We refer to these literals as "facts"

It could be understood that the conclusion has to be a simple literal in case (1), which is AFAIK not true. Maybe we could say:

> We refer to literals in these conclusions as "facts"